### PR TITLE
Prepare react-router-dom for v6 update

### PR DESCRIPTION
**Pull request by fiskus**
* Moved routes components from `component=` prop to `Route`'s children
* Use hooks instead of `RRDom.RouteComponentProps`
* Rewrite imports consistently: `import * as RRDom from...`

### CODE REVIEW

**Comment by nl0** [comment]
this one needs a merge/rebase

**Review by nl0 — COMMENTED** [review]
in general, this one looks ok. i'd say the new route notation is somewhat uglier, but whatever.
i think we should keep single imports as is to make this changeset somewhat smaller and use `RR` instead of `RRDom` for brevity

File: catalog/app/app.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import { createBrowserHistory as createHistory } from 'history'
 import * as React from 'react'
 import * as ReactDOM from 'react-dom'
-import { Router } from 'react-router-dom'
+import * as RRDom from 'react-router-dom'
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/app.tsx:7]: Single imports only? And wildcard if more than one?

**Comment by fiskus** [comment]
I think it worth looking at this again https://github.com/quiltdata/quilt/pull/1934/

File: catalog/app/app.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nl0) [reply catalog/app/app.tsx:7]: yeah, at least for the purpose of this refactoring. for new code we can use wildcard (namespace) imports consistently, but this probably should be left to the author's discretion

**Comment by nl0** [comment]
> I think it worth looking at this again #1934

feel free to revive that, but i don't see how it's relevant here. imports are already used more or less consistently. here i'm just trying to make the changeset somewhat smaller by proposing to keep single imports, since they are not that cumbersome in their current form.

**Comment by fiskus** [comment]
Reverted import changes

For `RR`, wildcard imports, and how coding styleguide PR is relevant here - let's start weekly 1-on-1's to resolve these issues.

File: catalog/app/containers/Admin/Admin.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react'
-import * as RR from 'react-router-dom'
+import * as RRDom from 'react-router-dom'
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/containers/Admin/Admin.tsx]: This is not strictly related to this PR, so let's do this in another PR with proper discussion

File: catalog/app/components/FileEditor/QuiltConfigEditor/WorkflowsToolbar.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import * as FP from 'fp-ts'
 import * as R from 'ramda'
 import * as React from 'react'
 import * as RF from 'react-final-form'
-import { useRouteMatch } from 'react-router-dom'
+import * as RRDom from 'react-router-dom'
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/components/FileEditor/QuiltConfigEditor/WorkflowsToolbar.tsx]: Reverted

File: catalog/app/components/BreadCrumbs/BreadCrumbs.spec.tsx [hunk 4]
```diff
@@ -40,7 +40,9 @@ describe('components/BreadCrumbs', () => {
       })
       const tree = renderer
         .create(
-          <MemoryRouter>{BreadCrumbs.render(crumbs, { getLinkProps })}</MemoryRouter>,
+          <RRDom.MemoryRouter>
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/components/BreadCrumbs/BreadCrumbs.spec.tsx]: reverted

**Review by nl0 — APPROVED** [review]

**Pull request merged**